protocol/xmrmaker/claim.go

```diff
@@ -65,15 +68,21 @@ func (s *swapState) claimFunds() (ethcommon.Hash, error) {
 		// TODO: eventually update when relayer discovery is implemented
 		txHash, err = s.claimRelayer()
 		if err != nil {
-			return ethcommon.Hash{}, err
+			log.Warnf("failed to claim using relayer at %s, trying relayers via p2p: err: %s",
+				s.offerExtra.RelayerEndpoint,
+				err,
+			)
+			txHash, err = s.discoverRelayersAndClaim()
 		}
+	} else if ethBalance.Uint64() == 0 {
+		txHash, err = s.discoverRelayersAndClaim()
```

Comment:
This logic is going to be difficult to document, but it is the type of information that end users will want to know. 

I'll try to describe what I think I'm reading. Let me know what I get right/wrong:
* To use a relayer, you should specify a non p2p http or https endpoint, or you should have an ethereum balance of exactly zero.
* If the specified non p2p relayer does not work, or if you did not specify a relayer and have a zero ethereum balance, all p2p relayers will be tried until one works.
* If you have a low ethereum balance and want to use a relayer, you must find one with an http or https endpoint and specify it when making the offer.

Reply:
yep that's right, also if you put some random string in `relayerEndpoint` it'll also fallback to discovering/trying relayers. I was thinking of adding another option inside the RPC `MakeOfferRequest` that's just a bool `useRelayer` so that the flow is more explicit (and maybe removing the HTTP endpoint, like I mentioned to you before), what do you think?

Reply:
Yeah, I don't see any advantage in supporting remote https relayers when everything else, besides localhost communication, is using libp2p.  If we only support libp2p relayers, would the flag be `--relayer-peer-id`?  And that flag would automatically set `--use-relayer` to `true` if it was not already set?  (Just tossing out ideas, I haven't thought about it for more than a minute.)

Reply:
yeah, currently relayer options are set per-offer, but I could see having the daemon have a `--use-relayer` flag be useful. and yeah we could change the endpoint to a libp2p endpoint (or array of endpoints)